onnxruntime/core/framework/cloud_invoker.cc

```diff
@@ -0,0 +1,260 @@
+#include "core/common/common.h"
+#include "core/framework/cloud_invoker.h"
+#include "core/framework/ort_value.h"
+
+namespace onnxruntime {
+
+namespace tc = triton::client;
+
+class TritonInvokder : public CloudEndPointInvoker {
+ public:
+  TritonInvokder(const CloudEndPointConfig& config);
+  void Send(gsl::span<const OrtValue> ort_inputs, std::vector<OrtValue>& ort_outputs) const noexcept override;
+
+ private:
+  static std::string MapDataType(int32_t ort_data_type);
+  std::string uri_;
+  std::string key_;  // access token for bearer authentication
+  std::string model_name_;
+  std::string model_ver_;
+  bool verbose_ = false;
+  onnxruntime::InlinedVector<std::string> input_names_;
+  onnxruntime::InlinedVector<std::string> output_names_;
+  std::shared_ptr<CPUAllocator> cpu_allocator_;
+  std::unique_ptr<triton::client::InferenceServerHttpClient> triton_client_;
+  tc::Headers http_headers_;
+};
+
+std::string TritonInvokder::MapDataType(int32_t ort_data_type) {
+  std::string triton_data_type;
+  switch (ort_data_type) {
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
+      triton_data_type = "FP32";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT8:
+      triton_data_type = "UINT8";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_INT8:
+      triton_data_type = "INT8";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT16:
+      triton_data_type = "UINT16";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_INT16:
+      triton_data_type = "INT16";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_INT32:
+      triton_data_type = "INT32";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_INT64:
+      triton_data_type = "INT64";
+      break;
+    //todo - do we need to support string?
+    //case ONNX_NAMESPACE::TensorProto_DataType_STRING:
+    //  triton_data_type = "BYTES";
+    //  break;
+    case ONNX_NAMESPACE::TensorProto_DataType_BOOL:
+      triton_data_type = "BOOL";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16:
+      triton_data_type = "FP16";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_DOUBLE:
+      triton_data_type = "FP64";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT32:
+      triton_data_type = "UINT32";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT64:
+      triton_data_type = "UINT64";
+      break;
+    case ONNX_NAMESPACE::TensorProto_DataType_BFLOAT16:
+      triton_data_type = "BF16";
+      break;
+    default:
+      break;
+  }
+  return triton_data_type;
+}
+
+TritonInvokder::TritonInvokder(const CloudEndPointConfig& config) : CloudEndPointInvoker(config) {
+  if (ReadConfig("uri", uri_) &&
+      ReadConfig("key", key_) &&
+      ReadConfig("model_name", model_name_) &&
+      ReadConfig("model_ver", model_ver_) &&
+      ReadConfig("input_names", input_names_) &&
+      ReadConfig("output_names", output_names_) &&
```

Comment:
For now we are expecting input and output names from customer configuration via session option. But should we infer them ourselves from the model that's been loaded?

Reply:
letting them being inferred.